lib/experimental/interactivity-api/class-wp-directive-processor.php

```diff
@@ -69,6 +69,44 @@ public static function has_root_block() {
 		return isset( self::$root_block );
 	}
 
+	/**
+	 * An array containing the main children of interactive.
+	 *
+	 * @var array
+	 */
+	public static $children_of_interactive_block = array();
+
+	/**
+	 * Add a root block to the variable.
+	 *
+	 * @param array $block The block to add.
+	 *
+	 * @return void
+	 */
+	public static function mark_children_of_interactive_block( $block ) {
+		self::$children_of_interactive_block[] = md5( serialize( $block ) );
+	}
+
+	/**
+	 * Remove a root block to the variable.
+	 *
+	 * @param array $block The block to remove.
+	 * @return void
+	 */
+	public static function unmark_children_of_interactive_block( $block ) {
+		self::$children_of_interactive_block = array_diff( self::$children_of_interactive_block, array( md5( serialize( $block ) ) ) );
```

Comment:
This removes all the occurrences instead of the first one, which can cause problems when there is more than one block with the same md5. I've changed it to remove only the first occurrence.

```suggestion
		$key = array_search( md5( serialize( $block ) ), self::$children_of_interactive_block, true );
		if ( false !== $key ) {
			unset( self::$children_of_interactive_block[ $key ] );
		}
```

Reply:
Done in https://github.com/WordPress/gutenberg/pull/56302/commits/276b64cef7aeb60be77f92a5277597174743a376.